api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImplTest.java

```diff
@@ -477,21 +478,94 @@ public void searchForObs_shouldSearchForObsByValueDate() {
 	}
 	
 	@Test
-	public void searchForObs_shouldSearchForObsByValueQuantity() {
+	public void searchForObs_shouldSearchForObsByValueQuantityWithPrefixAp() {
 		QuantityAndListParam quantityAndListParam = new QuantityAndListParam();
 		
 		QuantityOrListParam quantityOrListParam = new QuantityOrListParam();
 		
 		QuantityParam quantityParam = new QuantityParam();
-		quantityParam.setValue(VALUE_QUANTITY);
+		quantityParam.setValue("36");
+		quantityParam.setPrefix(ParamPrefixEnum.APPROXIMATE);
 		quantityAndListParam.addAnd(quantityOrListParam.add(quantityParam));
 		
 		Collection<Obs> results = dao.searchForObservations(null, null, null, null, null, quantityAndListParam, null, null,
 		    null, null);
 		
 		assertThat(results, notNullValue());
 		assertThat(results, not(empty()));
-		assertThat(results, hasItem(hasProperty("uuid", equalTo(OBS_VALUE_CONCEPT_UUID))));
+		assertThat(results, hasItem(hasProperty("uuid", equalTo("7e77d071-1f7d-4394-a316-0a458edf28c3"))));
+	}
+	
+	@Test
+	public void searchForObs_shouldSearchForObsByValueQuantityWithoutPrefixAndDecimalValue() {
+		QuantityAndListParam quantityAndListParam = new QuantityAndListParam();
+		
+		QuantityOrListParam quantityOrListParam = new QuantityOrListParam();
+		
+		QuantityParam quantityParam = new QuantityParam();
+		quantityParam.setValue("100.00");
+		quantityAndListParam.addAnd(quantityOrListParam.add(quantityParam));
+		
+		Collection<Obs> results = dao.searchForObservations(null, null, null, null, null, quantityAndListParam, null, null,
+		    null, null);
+		
+		assertThat(results, notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results, hasItem(hasProperty("uuid", equalTo("89fg071-1f7d-4394-a316-0a458edf28c3"))));
+	}
+	
+	//	@Test
+	//	public void searchForObs_shouldSearchForObsByValueQuantityWithoutPrefixAndEValue() {
+	//		QuantityAndListParam quantityAndListParam = new QuantityAndListParam();
+	//
+	//		QuantityOrListParam quantityOrListParam = new QuantityOrListParam();
+	//
+	//		QuantityParam quantityParam = new QuantityParam();
+	//		quantityParam.setValue("1e2");
+	//		quantityAndListParam.addAnd(quantityOrListParam.add(quantityParam));
+	//
+	//		Collection<Obs> results = dao.searchForObservations(null, null, null, null, null, quantityAndListParam, null, null,
+	//		    null, null);
+	//
+	//		assertThat(results, notNullValue());
+	//		assertThat(results, not(empty()));
+	//		assertThat(results, hasItem(hasProperty("uuid", equalTo("4efa62d2-6b8b-4803-a8fa-3f32ee54db4f"))));
```

Comment:
This has the wrong UUID, since we elected not to implement the approximation range without a decimal point, the search range for `1e2` is [99.5 ... 100.5], which is fine, but only UUID 89fg071-1f7d-4394-a316-0a458edf28c3 falls into that range. The other test works for me.

Reply:
But [here](https://www.hl7.org/fhir/search.html#number) I couldn't find the same range

[parameter]=1e2 | Values that equal 100, to 1 significant figures precision, so this is actually searching for values in the range [95 ... 105)
-- | --

Does 1e-2 works for you?

Reply:
I'm aware. This is a point where I'm suggesting we break spec, because the alternative seems to be to write a parser for scientific notation. Maybe we'll reassess later, but I don't think that's necessary at this point in time. And, yes, 1e-2 worked for me without modifications...